include/xsimd/types/xsimd_api.hpp

```diff
@@ -1033,6 +1033,14 @@ namespace xsimd
         return kernel::load_complex_aligned<A>(ptr, kernel::convert<batch_value_type> {}, A {});
     }
 
+#if XSIMD_ENABLE_XTL_COMPLEX
+    template <class To, class A = default_arch, class From, bool i3ec>
+    inline simd_return_type<xtl::xcomplex<From, From, i3ec>, To> load_as(xtl::xcomplex<From, From, i3ec> const* ptr, aligned_mode) noexcept
+    {
+        return load_as<To>(reinterpret_cast<std::complex<From> const*>(ptr), aligned_mode());
```

Comment:
Is this cast always valid? This looks very suspicious to me.

Reply:
The holy scriptures state that (§26.4): if a is an expression of type cv std::complex<T>* and the expression a[i] is well-defined for an integer expression i, then:

— reinterpret_cast<cv T*>(a)[2*i] shall designate the real part of a[i], and
— reinterpret_cast<cv T*>(a)[2*i + 1] shall designate the imaginary part of a[i].

Meaning the memory layout of a complex is [real_part, imag_part].

`xtl::complex` hs exactly the same layout, so this cast is safe. We already use it in the implementation of the load/store methods of a `batch<std::complex>` from a buffer of `xt::xcomplex` (see https://github.com/xtensor-stack/xsimd/blob/master/include/xsimd/types/xsimd_batch.hpp#L1184 for instance).